util/reggen/validate.py

```diff
@@ -1365,16 +1370,29 @@ def make_intr_regs(regs, offset, addrsep, fullwidth):
         return iregs, 1
 
     iregs.append(
-        make_intr_reg(regs, 'INTR_STATE', offset, 'rw1c', 'hrw',
-                      'Interrupt State Register'))
+        make_intr_alert_reg(regs, 'INTR_STATE', offset, 'rw1c', 'hrw',
+                            'Interrupt State Register'))
     iregs.append(
-        make_intr_reg(regs, 'INTR_ENABLE', offset + addrsep, 'rw', 'hro',
-                      'Interrupt Enable Register'))
+        make_intr_alert_reg(regs, 'INTR_ENABLE', offset + addrsep, 'rw', 'hro',
+                            'Interrupt Enable Register'))
     iregs.append(
-        make_intr_reg(regs, 'INTR_TEST', offset + 2 * addrsep, 'wo', 'hro',
-                      'Interrupt Test Register'))
+        make_intr_alert_reg(regs, 'INTR_TEST', offset + 2 * addrsep, 'wo', 'hro',
+                            'Interrupt Test Register'))
     return iregs, 0
 
+def make_alert_regs(regs, offset, addrsep, fullwidth):
+    alert_regs = []
+    alerts = regs['alert_list']
+    if sum([int(x['width'], 0) if 'width' in x else 1
+            for x in alerts]) > fullwidth:
+        log.error('More than ' + str(fullwidth) + ' alerts in list')
+        return alert_regs, 1
+
+    alert_regs.append(
+        make_intr_alert_reg(regs, 'ALERT_TEST', offset, 'wo', 'hro',
+                            'Alert Test Register'))
```

Comment:
Style nit: I'd consider writing this as
```python
    new_reg = make_intr_alert_reg(regs, 'ALERT_TEST', offset, 'wo', 'hro',
                                  'Alert Test Register')
    alert_regs.append(new_reg)
```
Same number of lines, but you get a variable name (admittedly not a particularly interesting one) and avoid the hanging indent.

Reply:
jup, rewritten (also above in the interrupt function).